scripts/launch/demo.sh

```diff
@@ -113,24 +121,39 @@ while [[ $i -lt $panesCnt ]]; do
   echo "Launching node $i in tab $im of window $ir"
   tmux select-pane -t $im
 
+  conf_file=$CONFIG
   wallet_args=''
   exec_name='cardano-node-simple'
-  if [[ $WALLET_TEST != "" ]]; then
-      if (( $i == $n - 1 )); then
-          wallet_args=" --tlscert $base/../tls-files/server.crt --tlskey $base/../tls-files/server.key --tlsca $base/../tls-files/ca.crt $wallet_flush" # --wallet-rebuild-db'
-          # We are using the different wallet exe here, depending on what was passed when the script was called.
-          exec_name=$WALLET_EXE_NAME
-          echo "Using wallet $exec_name."
-          if [[ $WALLET_DEBUG != "" ]]; then
-              wallet_args="$wallet_args --wallet-debug"
-          fi
+  if [[ $WALLET_TEST != "" ]] && [[ $i == $((n-1)) ]]; then
+      if [[ $WALLET_CONFIG != "" ]]; then
+          conf_file=$WALLET_CONFIG
+      fi
+      wallet_args=" --tlscert $base/../tls-files/server.crt --tlskey $base/../tls-files/server.key --tlsca $base/../tls-files/ca.crt $wallet_flush" # --wallet-rebuild-db'
+      wallet_args="$wallet_args --wallet-address 127.0.0.1:8090"
+      exec_name='cardano-node'
+      if [[ $WALLET_DEBUG != "" ]]; then
+          wallet_args="$wallet_args --wallet-debug"
       fi
   fi
-
   if [[ $i -lt $n ]]; then
-    node_cmd="$(node_cmd $i "$stats" "$wallet_args" "$system_start" "$config_dir" $exec_name) --no-ntp"
-    echo "$node_cmd"
-    tmux send-keys "$node_cmd" C-m
+    node_args="$(node_cmd $i "$stats" "$wallet_args" "$system_start" "$config_dir" "$conf_file")"
+    node_=$(find_binary $exec_name)
+    if [[ $WALLET_TEST != "" ]] && [[ $i == $((n-1)) ]]; then
+        updater_file="$config_dir/updater$i.sh"
+        launcher_=$(find_binary cardano-launcher)
+        launcher_args="--node $node_ "
+        if [[ "$UI" != "" ]]; then
+          ui_args_="$(prefix_args "$UI_ARGS" "-w")"
+          launcher_args=" $launcher_args --wallet $UI $ui_args_"
+        fi
+        node_args_="$(prefix_args "$node_args" "-n")"
+        launcher_args=" $launcher_args --updater /usr/bin/env -u bash --update-archive $updater_file"
+        launcher_args=" $launcher_args --node-timeout 5 $node_args_ --system-start 0 --wlogging --db-path $run_dir/node-db$i $rts_opts $reb $no_ntp $keys_args"
```

Comment:
`--wlogging` should be enabled only if `--wallet` is passed to the launcher (i. e. `"$UI" != ""`).

Reply:
If I do this then run the `us-test.sh` script it fails with 
`Invalid argument `scripts/launch/ui-simulator.sh'`

Reply:
What exactly did you try to do?
I think you should define `wlogging=""` before `if [[ "$UI" != "" ]]; then` and define `wlogging=" --wlogging "` inside that `if`.
Then you can change `--wlogging` to `$wlogging` on this line.

Reply:
Or even better, you can add `--wlogging` on line 147:
> launcher_args=" $launcher_args --wallet $UI $ui_args_"

becomes

> launcher_args=" $launcher_args --wlogging --wallet $UI $ui_args_"

Reply:
I did it also on line 147 but had the `--wallet` before  `--wlogging` flag and that is why it failed. All good now.